lib/autoupdate.ps1

```diff
@@ -318,34 +318,81 @@ function get_version_substitutions([String] $version, [Hashtable] $customMatches
     return $versionVariables
 }
 
-function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $matches) {
-    Write-Host -f DarkCyan "Autoupdating $app"
+function lets_do_autoupdates([String] $app, $json, [String] $version, $substitutions) {
+    $urls = @()
+    $hashes = @()
     $has_changes = $false
     $has_errors = $false
     [Bool]$valid = $true
-    $substitutions = get_version_substitutions $version $matches
 
-    if ($json.url) {
+    for ($i=0; $i -lt $json.autoupdate.url.Length; $i++) {
         # create new url
-        $url   = substitute $json.autoupdate.url $substitutions
-        $valid = $true
+        $url   = substitute $json.autoupdate.url[$i] $substitutions
+        $urls += $url
 
         if($valid) {
             # create hash
-            $hash = get_hash_for_app $app $json.autoupdate.hash $version $url $substitutions
+            if ($json.autoupdate.hash.length -gt $i) {
+                $h = $json.autoupdate.hash[$i]
+            } else {
```

Comment:
```suggestion
            } elseif ($json.autoupdate.hash.length -ne 0) {
                $h = $json.autoupdate.hash[-1]
            } else {
```

If `hash`'s length is less than `url`, the last `hash` expression is reused for remaining `url`s.

So we could use one `hash` expression for all these `url`s if hash extraction url and regex are the same, e.g., `$sha256.*?$basename`, different hashes for different `$basename`s.